overlord/devicestate/devicemgr.go

```diff
@@ -1453,6 +1441,39 @@ func (m *DeviceManager) runFDESetupHook(op string, params *boot.FDESetupHookPara
 	return hookResult, nil
 }
 
+func (m *DeviceManager) checkFDEFeatures(st *state.State) error {
+	// Run fde-setup hook with "op":"features". If the hook
+	// returns any {"features":[...]} reply we consider the
+	// hardware supported. If the hook errors or if it returns
+	// {"error":"hardware-unsupported"} we don't.
+	output, err := m.runFDESetupHook("features", &boot.FDESetupHookParams{})
+	if err != nil {
+		return err
+	}
+
+	// first check for an error
+	var errRes struct {
+		Error string `json:"error"`
+	}
+	if err := json.Unmarshal(output, &errRes); err == nil && errRes.Error != "" {
+		return fmt.Errorf("cannot use hook, it returned error: %v", errRes.Error)
+	}
+	// then check  for features, any feature reply is considered good
+	var featuresRes struct {
+		Features []string `json:"features"`
+	}
+	if err := json.Unmarshal(output, &featuresRes); err == nil {
```

Comment:
should we accept  {}  as valid and corresponding to {"features": []}?  I don't mind too much either way, atm we do but the tests don't check

Reply:
I think we should be strict initially, updated the code